ploggy/base/handler.py

```diff
@@ -6,6 +11,61 @@
 from .levels import Level, INFO
 
 
+def fake_function():
+    pass
+
+
+_srcfile = os.path.normcase(fake_function.__code__.co_filename)
+
+
+# Ref: https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L162
+if hasattr(sys, '_getframe'):
+    currentframe = lambda: sys._getframe(3)
+else: #pragma: no cover
+    def currentframe():
+        """Return the frame object for the caller's stack frame."""
+        try:
+            raise Exception
+        except Exception:
+            return sys.exc_info()[2].tb_frame.f_back
+
+
+def find_caller(stack_info=False, stacklevel=1):
+    """
+    Find the stack frame of the caller so that we can note the source
+    file name, line number and function name.
+
+    ref: https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L1553
+    """
+    f = currentframe()
+    if f is not None:
+        f = f.f_back
+    orig_f = f
+    while f and stacklevel > 1:
```

Comment:
not using stacklevel as of now. could potentially expose this from .log func in future.